src/workers/newView.ts

```diff
@@ -56,56 +56,63 @@ RETURNING "currentStreak", "maxStreak"
 const incrementReadingStreak = async (
   con: DataSource,
   data: DeepPartial<View>,
-): Promise<boolean> => {
-  const users = con.getRepository(User);
-  const repo = con.getRepository(UserStreak);
-  const { userId, timestamp } = data;
+): Promise<boolean> =>
+  con.transaction(async (entityManager) => {
```

Comment:
Why do we need a transaction here?

Reply:
I moved the alert update before the increment query, so I wanted this to rollback if for some reason the increment query failed